bots/jep/src/main/java/org/openjdk/skara/bots/jep/JEPBot.java

```diff
@@ -0,0 +1,141 @@
+
+    JEPBot(HostedRepository repo, IssueProject issueProject) {
+        this.repo = repo;
+        this.issueProject = issueProject;
+    }
+
+    @Override
+    public boolean concurrentWith(WorkItem other) {
+        if (!(other instanceof JEPBot otherBot)) {
+            return true;
+        }
+        return !repo.isSame(otherBot.repo);
+    }
+
+    @Override
+    public Collection<WorkItem> run(Path scratchPath) {
+        var prs = repo.pullRequests();
+        for (var pr : prs) {
+            var jepComment = pr.comments().stream()
+                    .filter(comment -> comment.author().equals(pr.repository().forge().currentUser()))
+                    .flatMap(comment -> comment.body().lines())
+                    .map(jepMarkerPattern::matcher)
+                    .filter(Matcher::find)
+                    .reduce((first, second) -> second)
+                    .orElse(null);
+            if (jepComment == null) {
+                if (pr.labelNames().contains(JEP_LABEL)) {
+                    pr.removeLabel(JEP_LABEL);
+                }
+                log.info("No jep command found in comment for PR-" + pr.id());
+                continue;
+            }
+
+            var issueId = jepComment.group(2);
+            if ("unneeded".equals(issueId)) {
+                if (pr.labelNames().contains(JEP_LABEL)) {
+                    pr.removeLabel(JEP_LABEL);
+                }
+                log.info("Found `/jep unneeded` command");
+                continue;
+            }
+
+            var issueOpt = issueProject.issue(issueId);
+            if (issueOpt.isEmpty()) {
+                log.severe("The issue `" + issueId + "` doesn't exist.");
+                continue;
+            }
+            var issue = issueOpt.get();
+
+            var issueType = issue.properties().get("issuetype");
+            if (issueType == null || !"JEP".equals(issueType.asString())) {
+                log.severe("The issue `" + issue.id() + "` is not a JEP.");
+                continue;
+            }
+
+            var issueStatus = issue.properties().get("status").get("name").asString();
+            var resolution = issue.properties().get("resolution");
+            String resolutionName = "";
+            if (resolution != null && !resolution.isNull() &&
+                    resolution.get("name") != null && !resolution.get("name").isNull()) {
+                resolutionName = resolution.get("name").asString();
+            }
+
+            var hasTargeted = "Targeted".equals(issueStatus) ||
+                    "Integrated".equals(issueStatus) ||
+                    "Completed".equals(issueStatus) ||
+                    ("Closed".equals(issueStatus) && "Delivered".equals(resolutionName));
+            if (hasTargeted && pr.labelNames().contains(JEP_LABEL)) {
+                pr.removeLabel(JEP_LABEL);
+            } else if (!hasTargeted && !pr.labelNames().contains(JEP_LABEL)) {
+                pr.addLabel(JEP_LABEL);
```

Comment:
Here we are taking action, I would like this logged at info level. Something like this:
```suggestion
                log.info("JEP issue " + issue.id() + " found in state " + issueStatus + ", removing JEP label from " + describe(pr);
                pr.removeLabel(JEP_LABEL);
            } else if (!hasTargeted && !pr.labelNames().contains(JEP_LABEL)) {
                log.info("JEP issue " + issue.id() + " found in state " + issueStatus + ", adding JEP label to " + describe(pr);
                pr.addLabel(JEP_LABEL);
```

Reply:
Fixed.